apps/dashr-lastodash/app.R

```diff
@@ -0,0 +1,379 @@
+    fig$x$layout[[xaxis]]$automargin <- TRUE
+    fig$x$layout[[xaxis]]$showline <- TRUE
+    fig$x$layout[[xaxis]]$title <- list(text = generate_axis_title(las_data$C, key), font = list(family = "Arial, sans-serif", size = font_size))
+    fig$x$layout[[xaxis]]$tickfont <- list(family = "Arial, sans-serif", size = tick_font_size)
+  }
+  # y-axis
+  fig$x$layout[["yaxis"]]$mirror <- "all"
+  fig$x$layout[["yaxis"]]$automargin <- TRUE
+  fig$x$layout[["yaxis"]]$showline <- TRUE
+  fig$x$layout[["yaxis"]]$title <- list(text = generate_axis_title(las_data$C, "DEPT"), font = list(family = "Arial, sans-serif", size = font_size))
+  fig$x$layout[["yaxis"]]$tickfont <- list(family = "Arial, sans-serif", size = tick_font_size)
+  fig$x$layout[["yaxis"]]$autorange <- 'reversed'
+  # layout
+  fig$x$layout$height <- height
+  fig$x$layout$width <- width
+  fig$x$layout$plot_bgcolor <- bg_color
+  fig$x$layout$paper_bgcolor <- bg_color
+  fig$x$layout$hovermode <- "y"
+  fig$x$layout$legend <- list("font" = list("size" = tick_font_size))
+  fig$x$layout$margin <- list(r = 0.0, t = 100.0, b = 50.0, l = 80.0, autoexpand = FALSE)
+  
+  return(dccGraph(figure = fig))
+}
+
+####################################################################################################
+# DEFINE APP LAYOUT
+
+app$layout(
+  htmlDiv(
+    list(
+      htmlDiv(
+        id = "controls", 
+        children = list(
+          htmlButton(
+            "Print", 
+            id = "las-print"
+          )
+        )
+      ),
+      htmlDiv(
+        id = "frontpage",
+        className = "page",
+        children = generate_frontpage()
+      ),
+      htmlDiv(
+        className = "section",
+        children = list(
+          htmlDiv(
+            className = "section-title",
+            children = "LAS well"
+          ),
+          htmlDiv(
+            className = "page",
+            children = list(
+              htmlDiv(
+                id = "las-table", 
+                children = generate_table()
+              ),
+              htmlDiv(
+                id = "las-table-print"
+              )
+            )
+          )
+        )
+      ),
+      htmlDiv(
+        className = "section",
+        children = list(
+          htmlDiv(
+            className = "section-title", 
+            children = "LAS curves"
+          ),
+          htmlDiv(
+            className = "page",
+            children = list(
+              htmlDiv(
+                id = "las-curves", 
+                children = generate_curves()
+              )
+            )
+          )
+        )
+      )
+    )
+  )
+)
+
+####################################################################################################
+# DEFINE APP CALLBACKS
+
+app$callback(
+  output = list(id = "las-table-print", property = "children"),
+  params = list(input(id = "table", property = "data")),
+  function (data) {
+    num_tables <- as.integer(length(data)/34)+1
```

Comment:
@ohaanika I think I understand why it makes sense here, but I'm always nervous about fixed numeric values being used to subset by rows or columns. Whenever possible, I try to use character strings (either to match on rows or columns, then subset by the index returned from something like `match`), since hardcoded indices tend to be fragile.

I see `34` reappear in the `start` and `ifelse` blocks below too. Just pointing it out for the future.